### DISCUSSION TITLE
Restrictions on data type sizes

### DISCUSSION THREAD

**Discussion by rnyoakum** [raceintospace/raceintospace#844] [Q&A]
It's been a while since I was immersed in the code, and I never did much with the Cereal updates, so this is aimed at @hweimer: is there a need for the Boost asserts guaranteeing the data structure sizes anymore, or are they antiquated (and removable)? It would be nice to change a few things like converting Astro names to strings from char[14].

**Comment by hweimer** [comment]
Good question. Concerning serialization of save files, it doesn't matter. So you won't break any save games if the checks at compile time are removed.

For the Data struct, however, there is also a runtime check whether Data->Checksum matches the size of the struct. This is only checked when loading the game data and not for save files. It detects when the binary and data files are belonging to different versions of the game, so it does have some value. On the other hand, if you want to replace a fixed size char array by a pointer-type class as std::string, this means that sizeof() will vary across different platforms, similar to what happened in #494.

In that case, I'd say the runtime check does more harm than good, so I'd support removing it as well.

**Reply by rnyoakum** [reply]
I checked into this a bit more - it seems the reason they're still around is support for the LegacyLoad function, which uses memory copying. Should LegacyLoad be blown away and the data structures freed from their tyranny? On one hand, I hate to just dump the legacy saves - though I'm not sure they still load in practice. On the other, it would make resolving #443, #350, and a few more pretty simple. Thoughts?
